Neos.Flow/Classes/Mvc/Controller/SimpleActionController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Neos\Flow\Mvc\Controller;
+
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Flow\Mvc\ActionResponse;
+use Neos\Flow\Mvc\Exception\NoSuchActionException;
+use Neos\Flow\SignalSlot\Exception\InvalidSlotException;
+
+/**
+ * Provides most direct access to our request/response abstraction,
+ * the request comes directly from the dispatcher and goes directly back to it.
+ *
+ * For helpers to facilitate throws, forwards, redirects:
+ * @see SpecialResponsesSupport
+ *
+ * Views or other processing needs to be added to your controller as needed,
+ * helpers will be suggested here as they become available.
+ */
+class SimpleActionController implements ControllerInterface
+{
+    /**
+     * @param ActionRequest $request
+     * @return ActionResponse
+     * @throws NoSuchActionException
+     * @throws InvalidSlotException
+     */
+    public function processRequest(ActionRequest $request): ActionResponse
+    {
+        $request->setDispatched(true);
+        $actionMethodName = $this->resolveActionMethodName($request);
+        return $this->$actionMethodName($request);
+    }
+
+    /**
+     * Resolves and checks the current action method name
+     *
+     * @return string Method name of the current action
+     * @throws NoSuchActionException
+     */
+    protected function resolveActionMethodName(ActionRequest $request): string
+    {
+        $actionMethodName = $request->getControllerActionName() . 'Action';
```

Comment:
do we want to keep the `*Action` pattern, because it still happens to me that i miss the suffix an it takes some time to figure out again what i missed 😅 , to fix this we could just improve the `routing:match` command or the like to emit a warning, or wait we cannot as this is highly dynamic and we have to have an actual request at hand.

Reply:
Mmm, yeah I thought about changing it, and I wouldn't mind that, for me I would actually say just make it an attribute? just allowing any method seems dangerous-ish especially if someone doesn't do proper sec policies. Also this then wouldn't match any existing policies as they all assume Action, so I just kept it that way.

Reply:
The ActionController will be part of https://github.com/neos/flow-development-collection/pull/3297